src/paths/steiner.cpp

```diff
@@ -0,0 +1,574 @@
+                IGRAPH_ALLOW_INTERRUPTION();
+                int_set F = fetchSetsBasedonIndex(holder - i, subsetMap);
+
+                int_set E;
+
+                std::set_difference(D.begin(), D.end(), F.begin(), F.end(), std::inserter(E, E.end()));
+
+                igraph_real_t temp_value = MATRIX(*dp_cache, *E.begin(), k) + MATRIX(*dp_cache, holder - i, k);
+
+                if (temp_value < min_value) {
+                    min_value = temp_value;
+                    min_E_value = *E.begin();
+                    min_F = F;
+                }
+            }
+
+            igraph_vector_int_t vectorlist_1;
+            IGRAPH_VECTOR_INT_INIT_FINALLY(&vectorlist_1, 0);
+
+            igraph_vector_int_t edgelist_1;
+            IGRAPH_VECTOR_INT_INIT_FINALLY(&edgelist_1, 0);
+
+            IGRAPH_CHECK(igraph_get_shortest_path_dijkstra(graph, &vectorlist_1, &edgelist_1, k, min_E_value, weights, IGRAPH_ALL));
+
+            IGRAPH_CHECK(igraph_vector_int_append(vectorlist_all, &vectorlist_1));
+            IGRAPH_CHECK(igraph_vector_int_append(edgelist_all, &edgelist_1));
+
+            igraph_vector_int_destroy(&vectorlist_1);
+            igraph_vector_int_destroy(&edgelist_1);
+            IGRAPH_FINALLY_CLEAN(2);
+        } else {
+            /*
+                If the size of subset is 2 then just shortest path from
+                k to first element of subset and k to second element of subset is sufficient.
+            */
+            igraph_integer_t E1, F1;
+
+            E1 = *D.begin();
+            F1 = *next(D.begin(), 1);
+
+
+            igraph_vector_int_t vectorlist_1;
+            IGRAPH_VECTOR_INT_INIT_FINALLY(&vectorlist_1, 0);
+
+            igraph_vector_int_t edgelist_1;
+            IGRAPH_VECTOR_INT_INIT_FINALLY(&edgelist_1, 0);
+
+            IGRAPH_CHECK(igraph_get_shortest_path_dijkstra(graph, &vectorlist_1, &edgelist_1, k, E1, weights, IGRAPH_ALL));
+
+            igraph_vector_int_t vectorlist_2;
+            IGRAPH_VECTOR_INT_INIT_FINALLY(&vectorlist_2, 0);
+
+            igraph_vector_int_t edgelist_2;
+            IGRAPH_VECTOR_INT_INIT_FINALLY(&edgelist_2, 0);
+
+            IGRAPH_CHECK(igraph_get_shortest_path_dijkstra(graph, &vectorlist_2, &edgelist_2, k, F1, weights, IGRAPH_ALL));
+
+            IGRAPH_CHECK(igraph_vector_int_append(vectorlist_all, &vectorlist_1));
+            IGRAPH_CHECK(igraph_vector_int_append(vectorlist_all, &vectorlist_2));
+
+            IGRAPH_CHECK(igraph_vector_int_append(edgelist_all, &edgelist_1));
+            IGRAPH_CHECK(igraph_vector_int_append(edgelist_all, &edgelist_2));
+
+            igraph_vector_int_destroy(&vectorlist_2);
+            igraph_vector_int_destroy(&vectorlist_1);
+
+            igraph_vector_int_destroy(&edgelist_1);
+            igraph_vector_int_destroy(&edgelist_2);
+
+            IGRAPH_FINALLY_CLEAN(4);
+
+            int_set min_F;
+            min_F.insert(F1);
```

Comment:
Here you create a new data structure, the modify it, but it immediately goes out of scope and gets destroyed.

Is there a mistake here?

Reply:
min_F is defined inside the while loop and it's used in if-else block. min_F is set to value D and D is the entry condition for the while loop which is why we have defined it.

Reply:
My point is that you defined two distinct variables named `min_F`. The one you are inserting into here is not the same as the one in the while loop.

Can you check this for mistakes?

Reply:
My apologies for the silly mistake. I have removed the code that re defines min_f inside the else block.